compositor_pipeline/src/pipeline/decoder/decoder_thread_video.rs

```diff
@@ -59,16 +64,29 @@ pub fn spawn_video_decoder_thread<Decoder: VideoDecoder, const BUFFER_SIZE: usiz
 
 fn init_decoder_stream<Decoder: VideoDecoder, const BUFFER_SIZE: usize>(
     ctx: Arc<PipelineCtx>,
+    extra_data: Option<Bytes>,
 ) -> Result<
     (
         impl Iterator<Item = PipelineEvent<Frame>>,
         DecoderThreadHandle,
     ),
     DecoderInitError,
 > {
+    let extra_data = extra_data
+        .map(ExtraData::parse)
+        .transpose()
+        .unwrap_or_else(|e| match e {
+            ExtraDataParseError::NotExtraData => None,
+            _ => {
+                warn!("Could not parse extra data: {e}");
+                None
+            }
+        });
+
     let (chunk_sender, chunk_receiver) = crossbeam_channel::bounded(BUFFER_SIZE);
-    let decoded_stream =
-        VideoDecoderStream::<Decoder, _>::new(ctx, chunk_receiver.into_iter())?.flatten();
+    // TODO: This should be only used if the stream is in h264
```

Comment:
You have 2 options here, either conditionally add that iterator or check the condition inside. First option would require boxing the resulting iterator, so it might be better to do it inside

Reply:
I went with the first option because it looks nicer imo